### PR TITLE
Rename and organize files

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
#1453 

## Description
This all started when I added a new file to Nos for getting media metadata. I spiraled into a furious tidying frenzy and just couldn't stop myself. I think this is better than what we had, but please let me know if you disagree.

## How to test
Build and run. Make sure the app is exactly the same as it was.

### CODE REVIEW

**Review by martindsq — CHANGES_REQUESTED** [review]
I like the overall idea. I'm ok with the vast majority of changes. But not some of them.

You are touching a lot, so it's hard to review them all. But given that I worked in some specific things, like note cards, it's easier to spot some of them. I'm going to mention the ones I found.

I assume `Components` is for shared views. What that said:

- `RepliesView` its not a component, but a screen. I'd like to see a new group alongside Discover, Home, Profile, etc.
- If we move  `RepliesView` to a new group, we should also move `ThreadView` inside that group because its used just by `RepliesView`.
- I also consider `MutesView` to be an independent screen,  but could be moved inside Profile I guess, as it is pushed by Profile alone.

I think that for future developers it could be helpful if we place a `README.md` inside some strategic folders explaining what we put inside. For example, `Components` is a candidate for sure. But I also would add `Fixtures` and `Modifiers` to that list. Nothing too complicated nor over-worked, just a quick explanation, like the kind we add when we comment on what a struct we added does, but for groups.

**Comment by martindsq** [comment]
Maybe doesn't belong here, but renaming `RepliesView` to something else more meaningful would be fine. It gives the impression that it shows a list of replies, but it does much more, and it makes it seem less important than it is.  A more meaningful name would be `NoteView` because its main responsibility is to display a note.

**Comment by martindsq** [comment]
Finally, you should create a ´devx´ ticket and mention it here.

**Comment by joshuatbrown** [comment]
> I like the overall idea. I'm ok with the vast majority of changes. But not some of them.

👍 

> I assume `Components` is for shared views. What that said:
> 
> * `RepliesView` its not a component, but a screen. I'd like to see a new group alongside Discover, Home, Profile, etc.

I disagree. Discover, Home, and Profile are all top-level navigation. Tap the Profile tab to go to the Profile screen. There's no tab, or top-level navigation (like our SideMenu), to get to the RepliesView. Additionally, you can get to the RepliesView in multiple ways, such as from the Profile, the Feed, or when you tap Reply on a note. To me, that makes it a component -- it can be reused in different places. My goal for the hierarchy here is that top-level navigation gets its own group.

> * If we move  `RepliesView` to a new group, we should also move `ThreadView` inside that group because its used just by `RepliesView`.

See above. I don't think RepliesView should get its own group, but I do agree that if it did, ThreadView should be inside that group.

> * I also consider `MutesView` to be an independent screen,  but could be moved inside Profile I guess, as it is pushed by Profile alone.

I can agree with this, and I'm fine with moving it inside Profile, so I'll do that.

> I think that for future developers it could be helpful if we place a `README.md` inside some strategic folders explaining what we put inside. For example, `Components` is a candidate for sure. But I also would add `Fixtures` and `Modifiers` to that list. Nothing too complicated nor over-worked, just a quick explanation, like the kind we add when we comment on what a struct we added does, but for groups.

I've never considered this, but I can think about it. We have plenty of examples of the types of files that go in each folder, and we have PR reviews where we can discuss whenever someone adds a new file, so I guess I'm not sure what benefit we'd get from it. Do you have an example of this somewhere?

> Maybe doesn't belong here, but renaming `RepliesView` to something else more meaningful would be fine. It gives the impression that it shows a list of replies, but it does much more, and it makes it seem less important than it is. A more meaningful name would be `NoteView` because its main responsibility is to display a note.

I agree, and I think this solves our disagreement above. I'll rename it `NoteView` and move it into `Components/Note`.

> Finally, you should create a `devx` ticket and mention it here.

I don't see the need for this, as it's not user-facing and it's a very small amount of effort. I thought we created `devx` tickets when we wanted to change bigger things and needed to prioritize the work. This was a small thing I did in my "30 minutes to do whatever needs to be done" time. But I might be wrong, and I'm willing to do it if we need to.

**Comment by martindsq** [comment]
> > * `RepliesView` its not a component, but a screen. I'd like to see a new group alongside Discover, Home, Profile, etc.
> 
> I disagree. Discover, Home, and Profile are all top-level navigation. Tap the Profile tab to go to the Profile screen. There's no tab, or top-level navigation (like our SideMenu), to get to the RepliesView. Additionally, you can get to the RepliesView in multiple ways, such as from the Profile, the Feed, or when you tap Reply on a note. To me, that makes it a component -- it can be reused in different places. My goal for the hierarchy here is that top-level navigation gets its own group.

I still disagree. When inspecting `RepliesView` I see that:

- It shows a navigation bar title because it calls `.nosNavigationBar(title: .localizable.thread)`
- It sends an analytics event marking that the user opened a specific screen with `analytics.showedThread()`

This makes me think that it is more of a screen than a component I can re-use in other places. Maybe we have different definitions of a component, for me it's a View meant to be reusable from other places.

> > I think that for future developers it could be helpful if we place a `README.md` inside some strategic folders explaining what we put inside. For example, `Components` is a candidate for sure. But I also would add `Fixtures` and `Modifiers` to that list. Nothing too complicated nor over-worked, just a quick explanation, like the kind we add when we comment on what a struct we added does, but for groups.
> 
> I've never considered this, but I can think about it. We have plenty of examples of the types of files that go in each folder, and we have PR reviews where we can discuss whenever someone adds a new file, so I guess I'm not sure what benefit we'd get from it. Do you have an example of this somewhere?
> 

We don't have an example of a README.md inside a folder other than the root one. I think `Components` deserve an explanation after we settle on a definition of what's goes inside. The benefit of this is that, when you are away, maybe on a vacation, other developers (maybe new to Nos) can read it when in doubt without having to consult you.

> > Finally, you should create a `devx` ticket and mention it here.
> 
> I don't see the need for this, as it's not user-facing and it's a very small amount of effort. I thought we created `devx` tickets when we wanted to change bigger things and needed to prioritize the work. This was a small thing I did in my "30 minutes to do whatever needs to be done" time. But I might be wrong, and I'm willing to do it if we need to.

I agree that it is not a strong need to do that. But when I did something similar in https://github.com/planetary-social/nos/pull/1383, without a user-facing impact, I was asked to create a ticket. I think we should aim to not make it arbitrary and look for a simple rule that makes PR reviewing the simplest as it could be. For me that would be, every PR should have a ticket associated. Otherwise, we should come up with a clear definition of what has a ticket and what not. A sort of rulebook for reviewing.

File: Nos/Views/Components/Note/NoteView.swift [hunk 1]
```diff
@@ -3,7 +3,7 @@ import SwiftUI
 import SwiftUINavigation
 import Dependencies
 
-struct RepliesView: View {
+struct NoteView: View {
```
Comment (martindsq) [inline Nos/Views/Components/Note/NoteView.swift:6]: 👍

**Review by martindsq — COMMENTED** [review]
The changes are fine. I just believe we should sort our theoretical discussion above before moving on.

Thanks for refactoring all this btw.

**Comment by joshuatbrown** [comment]
> > > * `RepliesView` its not a component, but a screen. I'd like to see a new group alongside Discover, Home, Profile, etc.
> > 
> > 
> > I disagree. Discover, Home, and Profile are all top-level navigation. Tap the Profile tab to go to the Profile screen. There's no tab, or top-level navigation (like our SideMenu), to get to the RepliesView. Additionally, you can get to the RepliesView in multiple ways, such as from the Profile, the Feed, or when you tap Reply on a note. To me, that makes it a component -- it can be reused in different places. My goal for the hierarchy here is that top-level navigation gets its own group.
> 
> I still disagree. When inspecting `RepliesView` I see that:
> 
> * It shows a navigation bar title because it calls `.nosNavigationBar(title: .localizable.thread)`
> * It sends an analytics event marking that the user opened a specific screen with `analytics.showedThread()`
> 
> This makes me think that it is more of a screen than a component I can re-use in other places. Maybe we have different definitions of a component, for me it's a View meant to be reusable from other places.

I guess I just left anything under Components that doesn't have top-level navigation in our app. The only things that have their own groups in Xcode are things that have their own tab in the app, or their own line in the Side Menu. For the tabs, we have groups for Home (Feed tab), Discover, NoteComposer (Post tab/button), Notifications, Profile, and then for the items in the Side Menu, we have Profile (Your Profile in the menu), Settings, and Relay (Relays in the menu). And AboutView is just part of the SideMenu group because it's so small and self-contained, though I suppose we could promote it to its own group with just one file.

But at this point I'm not sure it's worth continuing to debate, so I'll just move the `Note` group up a level and then I think we're good.

> > > I think that for future developers it could be helpful if we place a `README.md` inside some strategic folders explaining what we put inside. For example, `Components` is a candidate for sure. But I also would add `Fixtures` and `Modifiers` to that list. Nothing too complicated nor over-worked, just a quick explanation, like the kind we add when we comment on what a struct we added does, but for groups.
> > 
> > 
> > I've never considered this, but I can think about it. We have plenty of examples of the types of files that go in each folder, and we have PR reviews where we can discuss whenever someone adds a new file, so I guess I'm not sure what benefit we'd get from it. Do you have an example of this somewhere?
> 
> We don't have an example of a README.md inside a folder other than the root one. I think `Components` deserve an explanation after we settle on a definition of what's goes inside. The benefit of this is that, when you are away, maybe on a vacation, other developers (maybe new to Nos) can read it when in doubt without having to consult you.

I don't feel like anyone needs to consult me on this; it's up to all of us, as a team, to decide where to put things in our code. If someone adds a file to Components and you or Matt approves, that's fine, even if I would have suggested another place.

As for examples: I was asking if you have examples from another codebase, like another open source project or something.

> > > Finally, you should create a `devx` ticket and mention it here.
> > 
> > 
> > I don't see the need for this, as it's not user-facing and it's a very small amount of effort. I thought we created `devx` tickets when we wanted to change bigger things and needed to prioritize the work. This was a small thing I did in my "30 minutes to do whatever needs to be done" time. But I might be wrong, and I'm willing to do it if we need to.
> 
> I agree that it is not a strong need to do that. But when I did something similar in #1383, without a user-facing impact, I was asked to create a ticket. I think we should aim to not make it arbitrary and look for a simple rule that makes PR reviewing the simplest as it could be. For me that would be, every PR should have a ticket associated. Otherwise, we should come up with a clear definition of what has a ticket and what not. A sort of rulebook for reviewing.

I've created a ticket (#1453) to go along with this PR. The reason I thought #1383 needed a ticket was that it impacted reporting in PostHog -- a change that Linda and others might want to be aware of. This PR only affects developers because it only moves files to new locations (with a few renamed types). I know we merged #1419 without a ticket because it only corrected typos in comments and the CHANGELOG, and a ticket feels like a lot of process just to correct a few typos that don't affect the final product or any of our tooling or services (like PostHog). However, I understand the need for consistency, so maybe adding some extra work to create a tickte for every PR is OK.

**Comment by mplorentz** [comment]
Wow, I love that you guys are having thoughtful, nuanced discussion but I hate the way Github presents it. It is very hard to read. Maybe we should do PR comments in Nos (joke).

To weigh in on the "when do we need a ticket" question the current criteria is: if it changes anything non-developers will notice or it takes more than 30 minutes then it needs a ticket.

**Comment by martindsq** [comment]
> To weigh in on the "when do we need a ticket" question the current criteria is: if it changes anything non-developers will notice or it takes more than 30 minutes then it needs a ticket.

Agreed. I will open a PR that updates the PR template so that bit is clear.

**Comment by martindsq** [comment]
> But at this point I'm not sure it's worth continuing to debate, so I'll just move the `Note` group up a level and then I think we're good.

I apologize if our discussion felt overwhelming. For me, this refactor falls under the architecture umbrella, and I truly enjoy diving into software architecture as it’s always a great learning experience. I believe the changes you’ve proposed will have a positive impact in the long run. I’ll go ahead and approve this so we can keep moving forward.

**Review by martindsq — APPROVED** [review]

**Comment by joshuatbrown** [comment]
@martindsq thanks!

**Pull request merged**